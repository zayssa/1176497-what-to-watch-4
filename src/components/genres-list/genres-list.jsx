import React from "react";
import PropTypes from "prop-types";

export const GenresList = ({genres, currentGenre, setGenre}) => {
  return (
    <ul className="catalog__genres-list">
      {genres.map((genresOne) => (
        <li className={`catalog__genres-item ${genresOne === currentGenre ? `catalog__genres-item--active` : ``}`} key={`genres-${genresOne}`}>
          <a href="#" onClick={() => setGenre(genresOne)} className="catalog__genres-link">{genresOne}</a>
        </li>
      ))}
    </ul>
  );
};

GenresList.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  currentGenre: PropTypes.string.isRequired,
  setGenre: PropTypes.func.isRequired
};
