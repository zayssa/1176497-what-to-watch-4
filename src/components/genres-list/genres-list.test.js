import React from "react";
import renderer from "react-test-renderer";
import {GenresList} from "./genres-list.jsx";

it(`Render GenresList`, () => {
  const tree = renderer
    .create(<GenresList
      genres={[`comedy`, `horror`]}
      currentGenre="comedy"
      setGenre={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
