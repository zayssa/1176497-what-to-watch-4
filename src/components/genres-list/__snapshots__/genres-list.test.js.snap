// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render GenresList 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      comedy
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      horror
    </a>
  </li>
</ul>
`;
